app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/validation.ts

```diff
@@ -16,3 +16,9 @@ export const PREVIOUS_PASSWORD_PATTERN = '^[\\S]+.*[\\S]+$'
 export const PREVIOUS_PASSWORD_TITLE =
     'Your password must neither start nor end with whitespace, and must contain ' +
     'at least two characters.'
+
+/** Regex pattern for valid names for local projects. */
+export const LOCAL_PROJECT_NAME_PATTERN = '[A-Z][a-z]*(?:_\\d+|_[A-Z][a-z]*)*'
+/** Human readable explanation of project name restrictions for local projects. */
+export const LOCAL_PROJECT_NAME_TITLE =
+    'Project names must be in Upper_Snake_Case. (Numbers (_0, _1) are also allowed.)'
```

Comment:
What are "local projects"? If they are "not cloud projects", why we have different validation for local and cloud project names?

Reply:
cloud projects (currently) have no restrictions on project name. local projects have a restriction put in place by the Project Manager.

Reply:
comment added. (the reason why cloud projects don't have the same restriction is because the name in the database is not related to the name passed to the cloud VM's Project Manager - which i believe is always `Project_root`)